src/icrc1_auction_api.mo

```diff
@@ -571,6 +604,44 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
     await* btcHandler.getWithdrawalStatus(arg);
   };
 
+  public shared ({ caller }) func cycles_withdraw(args : { to : Principal; amount : Nat }) : async DirectCyclesWithdrawResult {
+    let ?cyclesAssetId = getAssetId(TCYCLES_LEDGER_PRINCIPAL) else throw Error.reject("Cycles asset is not supported");
+
+    let ledgerFee = 100_000_000;
+    if (args.amount <= ledgerFee) {
+      return #Err(#TooLowAmount {});
+    };
+
+    let (rollbackCredit, doneCallback) = switch (auction.deductCredit(caller, cyclesAssetId, args.amount)) {
+      case (#err _) return #Err(#InsufficientCredit({}));
+      case (#ok(_, r, d)) (r, d);
+    };
+    let ledger : (
+      actor {
+        withdraw : shared ({
+          to : Principal;
+          from_subaccount : ?[Nat8];
+          created_at_time : ?Nat64;
+          amount : Nat;
+        }) -> async ({
+          #Ok : Nat;
+          #Err : CyclesLedgerWithdrawError;
+        });
+      }
+    ) = actor (Principal.toText(TCYCLES_LEDGER_PRINCIPAL));
```

Comment:
Can this be outside of the `cycles_withdraw` function, assigned in top-level?

Reply:
done